openvdb/points/PointConversion.h

```diff
@@ -490,27 +501,15 @@ struct ConvertPointDataGridPositionOp {
 
             if (leaf.pos() > 0) offset += mPointOffsets[leaf.pos() - 1];
 
-            auto handle = AttributeHandle<ValueType>::create(leaf->constAttributeArray(mIndex));
-
-            if (useGroups) {
-                auto iter = leaf->beginIndexOn(MultiGroupFilter(mIncludeGroups, mExcludeGroups, leaf->attributeSet()));
+            auto handle = SourceHandleT::create(leaf->constAttributeArray(mIndex));
 
-                for (; iter; ++iter) {
-                    const Vec3d xyz = iter.getCoord().asVec3d();
-                    const Vec3d pos = handle->get(*iter);
-                    pHandle.set(static_cast<Index>(offset++), /*stride=*/0,
-                        mTransform.indexToWorld(pos + xyz));
-                }
+            if (mFilter.state() == index::ALL) {
+                auto iter = leaf->beginIndexOn();
+                convert(iter, pHandle, *handle, offset);
             }
             else {
-                auto iter = leaf->beginIndexOn();
-
-                for (; iter; ++iter) {
-                    const Vec3d xyz = iter.getCoord().asVec3d();
-                    const Vec3d pos = handle->get(*iter);
-                    pHandle.set(static_cast<Index>(offset++), /*stride=*/0,
-                        mTransform.indexToWorld(pos + xyz));
-                }
+                auto iter = leaf->beginIndexOn(newFilter);
```

Comment:
beginIndexOn will again copy the filter... tbh i think we should remove this internal copying of the filter in beginIndex() but if not, move the copying into the required branches

Reply:
This needs to be updated throughout this file

Reply:
You're right, I'll address this.